u8timeseries/custom_logging.py

```diff
@@ -0,0 +1,71 @@
+import sys
+import logging
+from datetime import datetime
+import pathlib
+import time
+
+def get_logger(name):
+    """
+    Internally calls the logging.getLogger function with the 'name' argument to create or 
+    retrieve a logger object. It is recommended to pass __name__ as argument when calling 
+    get_logger. The returned logger object logs to the standard error stream and formats
+    the messages appropriately.
+
+    :param name: The name that gets passed to the logger.getLogger function.
+    :return: A logger instance with the given name.
+    """
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.INFO)
+    stderr_handler = logging.StreamHandler()
+    formatter = logging.Formatter('[%(asctime)s] %(levelname)s | %(name)s | %(message)s')
+    stderr_handler.setFormatter(formatter)
+    logger.addHandler(stderr_handler)
+    return logger
+
+def raise_if_not(boolean_value: bool, message: str = "", logger: logging.Logger = get_logger('main_logger')):
+    """
+    Checks provided boolean expression and raises a ValueError if it evaluates to False.
+    It logs the error to the provided logger before raising it.
+
+    :param boolean_value: The boolean expression (or value) to be checked.
```

Comment:
I'm nitpicking here, but instead of "boolean expression (or value)" you can use "predicate", as this is the term for it :)

Reply:
Should I also change the name of the variable 'boolean_value'?

Reply:
In fact I just googled it and a predicate is a function returning a boolean. Here we do have a boolean value everywhere in fact, so maybe you could just rename it "condition" everywhere? :D

Reply:
Ok will do!

Reply:
done